lib/main.js

```diff
@@ -420,23 +418,6 @@ const Hydrogen = {
     }
   },
 
-  restartKernelAndReEvaluateBubbles() {
-    const { editor, kernel, markers } = store;
-    if (!editor || !kernel || !markers) return;
-
-    let breakpoints = [];
-    markers.markers.forEach((bubble: ResultView) => {
-      breakpoints.push(bubble.marker.getBufferRange().start);
-    });
-    markers.clear();
-
-    if (!editor || !kernel) {
-      this.runAll(breakpoints);
-    } else {
-      kernel.restart(() => this.runAll(breakpoints));
-    }
-  },
-
   toggleBubble() {
```

Comment:
I believe `toggleBubble` is only used for the command, thus you should drop this as well.
(Plz make a double check)

Reply:
`toggleBubble` is a separate feature

Reply:
Ah really ? Plz explain the usage.

Reply:
Refer to this: https://nteract.gitbooks.io/hydrogen/docs/Usage/GettingStarted.html#hydrogen-toggle-bubble

Reply:
It was also used for clearing bubbles singularly and not all at once. I have temporarily refactored toggleBubble to clearBubble which could be used to remove one bubble or many within a selection which was a feature of toggleBubble